base/linalg/arnoldi.jl

```diff
@@ -349,23 +349,36 @@ Lanczos tridiagonalization on the Hermitian matrix
 plus and minus the singular values of ``A``.
 """
 svds(A; kwargs...) = _svds(A; kwargs...)
-function _svds(X; nsv::Int = 6, ritzvec::Bool = true, tol::Float64 = 0.0, maxiter::Int = 1000)
+function _svds(X; nsv::Int = 6, ritzvec::Bool = true, tol::Float64 = 0.0, maxiter::Int = 1000, ncv::Int = 2*nsv, v0::Vector=zeros(eltype(X),(0,)))
     if nsv < 1
         throw(ArgumentError("number of singular values (nsv) must be ≥ 1, got $nsv"))
     end
     if nsv > minimum(size(X))
         throw(ArgumentError("number of singular values (nsv) must be ≤ $(minimum(size(X))), got $nsv"))
     end
+    m,n = size(X)
     otype = eltype(X)
-    ex    = eigs(SVDOperator(X), I; ritzvec = ritzvec, nev = 2*nsv, tol = tol, maxiter = maxiter)
-    ind   = [1:2:nsv*2;]
+    if length(v0) == n
```

Comment:
It's ambiguous which way the user would like to initialize then. Would it make sense to be strict and require complete initialization, or provide some different mechanism to indicate the intent of which part of the vector should be initialized with the user provided values?